### PR TITLE
ci: use GCS cache for vcpkg on Kokoro+Windows

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #9591

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9878)

### CODE REVIEW

File: ci/kokoro/windows/build-cmake-dependencies.ps1 [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 # Stop on errors. This is similar to `set -e` on Unix shells.
 $ErrorActionPreference = "Stop"
+. ci/kokoro/windows/lib/vcpkg-cache.ps1
```
Comment (devbww) [inline ci/kokoro/windows/build-cmake-dependencies.ps1:19]: Perhaps these lines deserve to be offset from the "Stop on errors" block?

File: ci/kokoro/windows/lib/vcpkg-cache.ps1 [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Helper functions to configure the vcpkg cache
+
+function Configure-Vcpkg-Cache {
+    if (-not (Test-Path env:KOKORO_GFILE_DIR)) {
+        Write-Host -ForegroundColor Yellow "`n$(Get-Date -Format o) " `
+            "Will not configure a remove vcpkg cache as KOKORO_GFILE_DIR is not set."
```
Comment (dbolduc) [inline ci/kokoro/windows/lib/vcpkg-cache.ps1:20]: s/remove/remote/

**Review by devbww — APPROVED** [review]

File: ci/kokoro/windows/lib/vcpkg-cache.ps1 [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        return
+    }
+    $CACHE_KEYFILE="${env:KOKORO_GFILE_DIR}/build-results-service-account.json"
+    &gcloud auth activate-service-account --key-file ${CACHE_KEYFILE}
+    if ($LastExitCode) {
+        Write-Host -ForegroundColor Yellow "`n$(Get-Date -Format o) " `
+            "Will not configure a remove vcpkg cache as the service account activation failed."
```
Comment (dbolduc) [inline ci/kokoro/windows/lib/vcpkg-cache.ps1:27]: s/remove/remote/

File: ci/kokoro/windows/lib/vcpkg-cache.ps1 [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    } else {
+        $CACHE_BUCKET="cloud-cpp-kokoro-results"
+    }
+
+    if (Test-Path env:MSVC_VERSION) {
+        $CACHE_FOLDER="${CACHE_BUCKET}/build-cache/google-cloud-cpp/vcpkg/windows/main/${env:MSVC_VERSION}/"
+    } else {
+        $CACHE_FOLDER="${CACHE_BUCKET}/build-cache/google-cloud-cpp/vcpkg/windows/main/default/"
+    }
+    ${env:VCPKG_BINARY_SOURCES}="x-gcs,gs://${CACHE_BUCKET}/${CACHE_FOLDER},readwrite"
```
Comment (dbolduc) [inline ci/kokoro/windows/lib/vcpkg-cache.ps1:42]: it looks like `$CACHE_FOLDER` already starts with `"$CACHE_BUCKET/"`. Should we get rid of the leading `"${CACHE_BUCKET}/"` ?

Comment (coryan) [reply ci/kokoro/windows/lib/vcpkg-cache.ps1:42]: I noticed and for some reason I decided to wait and fix Windows and macOS at the same time.  But fixing now is better.  Also created #9879 for the macOS cleanup.

File: ci/kokoro/windows/lib/vcpkg-cache.ps1 (hunk 3 above)
Comment (coryan) [reply ci/kokoro/windows/lib/vcpkg-cache.ps1]: Fixed.

File: ci/kokoro/windows/lib/vcpkg-cache.ps1 (hunk 2 above)
Comment (coryan) [reply ci/kokoro/windows/lib/vcpkg-cache.ps1]: Fixes

File: ci/kokoro/windows/build-cmake-dependencies.ps1 (hunk 1 above)
Comment (coryan) [reply ci/kokoro/windows/build-cmake-dependencies.ps1:21]: Done.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**